src/orion/core/worker/experiment.py

```diff
@@ -133,22 +140,37 @@ class Experiment:
     )
     non_branching_attrs = ("max_trials", "max_broken")
 
-    def __init__(self, name, version=None, mode="r"):
-        self._id = None
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        name: str,
+        space: Space | dict[str, str],
+        version: int | None = 1,
+        mode: Mode = "r",
+        _id: str | int | None = None,
+        max_trials: int | None = None,
+        max_broken: int | None = None,
+        algorithms: BaseAlgorithm | None = None,
+        working_dir: str | None = None,
+        metadata: dict | None = None,
+        refers: dict | None = None,
+    ):
+        self._id = _id
         self.name = name
+        self.space: Space = (
```

Comment:
Where is the experiment created with a space config instead of with the space object?

Reply:
Nowhere atm. I can remove this if you want.

Reply:
Removed in 6da0bc9c6ab1c1f11948846551a8b09e9ff0eef6